src/hooks/protections/receive_net_message.cpp

```diff
@@ -98,227 +324,439 @@ namespace big
+				return true;
+			}
 
-				if (player->is_spammer)
+			if (player)
+			{
+				if (player->block_radio_requests)
 					return true;
 
-				if (auto spam_reason = chat::is_text_spam(message, player))
+				if (player->m_radio_request_rate_limit.process())
 				{
-					if (g.session.log_chat_messages)
-						chat::log_chat(message, player, spam_reason, is_team);
-					
-					player->is_spammer = true;
-					if (!(player->is_trusted || (player->is_friend() && g.session.trust_friends) || g.session.trust_session))
+					if (player->m_radio_request_rate_limit.exceeded_last_process())
 					{
-						session::add_infraction(player, Infraction::CHAT_SPAM);
-						g.reactions.chat_spam.process(player);
+						session::add_infraction(player, Infraction::TRIED_KICK_PLAYER);
+						g_notification_service.push_error("PROTECTIONS"_T.data(),
+						    std::vformat("OOM_KICK"_T, std::make_format_args(player->get_name())));
+						player->block_radio_requests = true;
 					}
 					return true;
 				}
-				else
-				{
-					if (g.session.log_chat_messages)
-						chat::log_chat(message, player, SpamReason::NOT_A_SPAMMER, is_team);
-					if (g.session.chat_translator.enabled)
-					{
-						chat_message new_message{player->get_name(), message};
-						translate_queue.push(new_message);
-					}
-
-					if (g.session.chat_commands && message[0] == g.session.chat_command_prefix)
-						command::process(std::string(message + 1), std::make_shared<chat_command_context>(player));
-					else
-						g_lua_manager->trigger_event<menu_event::ChatMessageReceived>(player->id(), message);
-
-					if (msgType == rage::eNetMessage::MsgTextMessage && g_pointers->m_gta.m_chat_data && player->get_net_data())
-					{
-						buffer.Seek(0);
-						return g_hooking->get_original<hooks::receive_net_message>()(netConnectionManager, a2, frame); // Call original function since we can't seem to handle it
-					}
-				}
-				break;
 			}
-			case rage::eNetMessage::MsgScriptMigrateHost:
+			else
 			{
-				if (player->m_host_migration_rate_limit.process())
+				static rate_limiter unk_player_radio_requests{1s, 5};
+
+				if (unk_player_radio_requests.process())
 				{
-					if (player->m_host_migration_rate_limit.exceeded_last_process())
+					if (unk_player_radio_requests.exceeded_last_process())
 					{
-						session::add_infraction(player, Infraction::TRIED_KICK_PLAYER);
-						auto p_name = player->get_name();
-
-						g_notification_service.push_error("PROTECTIONS"_T.data(), std::vformat("OOM_KICK"_T, std::make_format_args(p_name)));
+						g_notification_service.push_error("PROTECTIONS"_T.data(), std::vformat("OOM_KICK"_T.data(), std::make_format_args(peer->m_info.name)));
 					}
 					return true;
 				}
+			}
+			break;
+		}
+		case rage::eNetMessage::MsgRadioStationSync:
+		{
+			if (!session || is_host_of_session(session, event->m_peer_id))
+			{
+				LOGF(stream::net_messages, WARNING, "{} sent MsgRadioStationSync, but is not the host", peer->m_info.name);
```

Comment:
shouldnt be it `!is_host_of_session(session, event->m_peer_id)` ? The not operator!